src/toil/cwl/cwltoil.py

```diff
@@ -3285,12 +3297,46 @@ def generate_default_job_store(
     ]
 )
 
+def format_output(output: Any) -> None:
+    """
+    Change the output object to match cwltool
+    """
+    if not isinstance(output, dict):
+        return
+    def change_output(output: Union[List[Any], Dict[Any, Any]]) -> None:
+        if isinstance(output, list):
+            for nested_output in output:
+                change_output(nested_output)
+        if isinstance(output, dict) and output.get("class") in ("File", "Directory"):
+            # Add 'path' with unquoted filepath
+            if output["location"].startswith("file://"):
+                if output.get("path") is None:
+                    output["path"] = uri_file_path(output["location"])
+                # Else if not a file, then don't decode and don't put in path
+                # Remove 'nameroot' and 'nameext'
+                if output.get("nameroot") is not None:
+                    del output["nameroot"]
+                if output.get("nameext") is not None:
+                    del output["nameext"]
+            # Recurse through nested objects
+            if output.get("listing") is not None:
+                for nested_object in output["listing"]:
+                    change_output(nested_object)
+    # Both the 'result' and 'log' output objects need to be changed
+    if output.get("result") is not None:
```

Comment:
I originally thought the issue the tests weren't passing was because something about the format of the output was incorrect. Looking at the compare function in cwltest though, I think the real reason why the tests was failing was because [the path it was grabbing](https://github.com/common-workflow-language/cwltest/blob/1987ca25d76975e41974bea3a34591051ece6379/cwltest/compare.py#L165C32-L165C32) was URL encoded and thus did not exist. I think the proper fix is to add the `path` field too when `location` is added during the file staging step, and I'll remove this function as it's excessive/unnecessary.